simulatingrisk/hawkdove/model.py

```diff
@@ -0,0 +1,155 @@
+from enum import Enum
+import math
+
+import mesa
+
+from simulatingrisk.utils import coinflip
+
+Play = Enum("Play", ["HAWK", "DOVE"])
+play_choices = [Play.HAWK, Play.DOVE]
+
+
+# divergent color scheme, nine colors
+# from https://colorbrewer2.org/?type=diverging&scheme=RdYlGn&n=9
+divergent_colors_9 = [
+    "#d73027",
+    "#f46d43",
+    "#fdae61",
+    "#fee08b",
+    "#ffffbf",
+    "#d9ef8b",
+    "#a6d96a",
+    "#66bd63",
+    "#1a9850",
+]
+
+# divergent color scheme, ficolors
+# from https://colorbrewer2.org/?type=diverging&scheme=RdYlGn&n=5
+divergent_colors_5 = ["#d7191c", "#fdae61", "#ffffbf", "#a6d96a", "#1a9641"]
+
+
+class HawkDoveAgent(mesa.Agent):
+    """
+    An agent with a risk attitude playing Hawk or Dove
+    """
+
+    def __init__(self, unique_id, model):
+        super().__init__(unique_id, model)
+
+        self.points = 0
+        self.choice = self.initial_choice()
+        self.last_choice = None
+
+        # risk level
+        # - based partially on neighborhood size,
+        #  which is configurable at the model level
+        num_neighbors = 8 if self.model.include_diagonals else 4
+        self.risk_level = self.random.randint(0, num_neighbors)
+
+    def initial_choice(self):
+        # first round : choose what to play randomly or based on initial setup
+        return coinflip(play_choices)
+
+    @property
+    def neighbors(self):
+        # use configured neighborhood (with or without diagonals) on the model;
+        # don't include the current agent
+        return self.model.grid.get_neighbors(
+            self.pos, moore=self.model.include_diagonals, include_center=False
+        )
+
+    @property
+    def num_hawk_neighbors(self):
+        """count how many neighbors played HAWK on the last round"""
+        return len([n for n in self.neighbors if n.last_choice == Play.HAWK])
+
+    def choose(self):
+        "decide what to play this round"
+        # after the first round, choose based on what neighbors did last time
+        if self.model.schedule.steps > 0:
+            # store previous  choice
+            self.last_choice = self.choice
+
+            # TODO: how to make risk attitude consistent with other sims?
+            # agent with r = 0 should always take the risky choice
+            #   (any risk is acceptable).
+            # agent with r = max should always take the safe option
+            #   (no risk is acceptable)
+            if self.risk_level < self.num_hawk_neighbors:
+                self.choice = Play.HAWK
+            else:
+                self.choice = Play.DOVE
+
```

Comment:
@LaraBuchak I've got a preliminary implementation of the hawk/dove game implemented but I'm having trouble figuring out the right way to code the logic for an agent's choice so that we define risk attitudes similar to what we've done in other simulations. 

What you wrote initially was:
> Let `h` be the number of neighbors who played HAWK during the previous round. If `h > r`, then play DOVE. Otherwise play HAWK.

... but we had a note about checking how we're doing this comparison.

That implementation doesn't seem consistent with how we're doing risk attitudes in the other games.   I've switched it in the code above, but I don't think what I've done is actually logical when you look at the numbers.  I'm wondering if it would be easier to think in terms of percentages so we can define risk attitudes the same way we have elsewhere.